components/script/dom/htmlvideoelement.rs

```diff
@@ -100,20 +102,54 @@ impl HTMLVideoElement {
         )
     }
 
-    pub fn get_video_width(&self) -> u32 {
-        self.video_width.get()
-    }
+    pub fn resize(&self, width: u32, height: u32) {
+        if self.video_width.get() == width && self.video_height.get() == height {
+            return;
+        }
 
-    pub fn set_video_width(&self, width: u32) {
         self.video_width.set(width);
+        self.video_height.set(height);
+
+        let window = window_from_node(self);
+        let task_source = window.task_manager().media_element_task_source();
+        task_source.queue_simple_event(self.upcast(), atom!("resize"), &window);
     }
 
-    pub fn get_video_height(&self) -> u32 {
-        self.video_height.get()
+    fn get_length(&self, name: &LocalName) -> Option<u32> {
+        self.upcast::<Element>()
+            .get_attribute(&ns!(), name)
+            .and_then(|attr| {
+                let LengthOrPercentageOrAuto::Length(len) = parse_length(&attr.value()) else {
+                    return None;
+                };
+                Some(len.to_px() as u32)
+            })
     }
 
-    pub fn set_video_height(&self, height: u32) {
-        self.video_height.set(height);
+    /// <https://html.spec.whatwg.org/multipage/#concept-video-intrinsic-width>
+    fn set_intrinsic_size(&self) {
+        let named_width = self.get_length(&local_name!("width"));
+        let named_height = self.get_length(&local_name!("height"));
+
+        // Default width is twice the height
+        let width = if let Some(x) = named_width {
+            x
+        } else if let Some(x) = named_height {
+            x * 2
+        } else {
+            DEFAULT_WIDTH
+        };
+
+        // Default height is half the width
+        let height = if let Some(x) = named_height {
+            x
+        } else if let Some(x) = named_width {
+            x / 2
+        } else {
+            DEFAULT_HEIGHT
+        };
```

Comment:
I tend to think this logic should actually be handled by https://github.com/servo/servo/blob/e857cdf022f319ae3fbd54e10e279ab43485995b/components/layout_2020/replaced.rs#L416